libpam/pam_handlers.c

```diff
@@ -17,7 +17,7 @@
 #include <fcntl.h>
 #include <unistd.h>
 
-#define BUF_SIZE                  1024
+#define BUF_SIZE                  4096
```

Comment:
Since you're introducing a dynamic memory allocation, what's the reason of changing default?